src/utils/common.ts

```diff
@@ -283,3 +283,83 @@ export const removeHoveredClassFromMonth = (month: string) => {
     });
   }
 };
+
+export const syncUpdatedProjectFinancesWithStartYear = (finances: IProjectFinances, startYear: number) => {
+  let convertedFinances: IProjectFinances = {
+    year: startYear,
+    budgetProposalCurrentYearPlus0: "0.00",
+    budgetProposalCurrentYearPlus1: "0.00",
+    budgetProposalCurrentYearPlus2: "0.00",
+    preliminaryCurrentYearPlus3: "0.00",
+    preliminaryCurrentYearPlus4: "0.00",
+    preliminaryCurrentYearPlus5: "0.00",
+    preliminaryCurrentYearPlus6: "0.00",
+    preliminaryCurrentYearPlus7: "0.00",
+    preliminaryCurrentYearPlus8: "0.00",
+    preliminaryCurrentYearPlus9: "0.00",
+    preliminaryCurrentYearPlus10: "0.00"
+  }
+
+  const yearDifference = finances.year - startYear;
+
+  for (const key in finances) {
+    if (key != "year") {
+      const num =  parseInt(key.replace(/\D/g, ""));
+      const convertedNumber = num + yearDifference;
+      if (convertedNumber >= 0 && convertedNumber < 3) {
+        const convertedKey = ("budgetProposalCurrentYearPlus" + convertedNumber) as keyof IProjectFinances;
+        const updatedFinances = {
+          [convertedKey]: finances[key as keyof IProjectFinances]
+        };
+        convertedFinances = { ...convertedFinances, ...updatedFinances };
+      } else if (convertedNumber <= 10) {
+        const convertedKey = ("preliminaryCurrentYearPlus" + convertedNumber) as keyof IProjectFinances;
+        const updatedFinances = {
+          [convertedKey]: finances[key as keyof IProjectFinances]
+        };
+        convertedFinances = { ...convertedFinances, ...updatedFinances };
+      }
+    }
+  }
+  return convertedFinances;
+}
+
+export const syncUpdatedClassFinancesWithStartYear = (finances: IClassFinances, startYear: number) => {
+  const initialBudgets = {
+    plannedBudget: 0.00,
+    frameBudget: 0.00,
+    isFrameBudgetOverlap: false
+  }
+  let convertedFinances: IClassFinances = {
+    ...finances,
+    year: startYear,
+    year0: initialBudgets,
+    year1: initialBudgets,
+    year2: initialBudgets,
+    year3: initialBudgets,
+    year4: initialBudgets,
+    year5: initialBudgets,
+    year6: initialBudgets,
+    year7: initialBudgets,
+    year8: initialBudgets,
+    year9: initialBudgets,
+    year10: initialBudgets
+  }
+
+  const yearDifference = finances.year - startYear;
+
+  for (const key in finances) {
+    if (!["year", "budgetOverrunAmount", "projectBudgets"].includes(key)) {
+      const num =  parseInt(key.replace(/\D/g, ""));
+      const convertedNumber = num + yearDifference;
+      if (convertedNumber >= 0 && convertedNumber <= 10) {
+        const convertedKey = ("year" + convertedNumber) as keyof IClassFinances;
+        const updatedFinances = {
+          [convertedKey]: finances[key as keyof IClassFinances]
+        };
+        convertedFinances = { ...convertedFinances, ...updatedFinances };
+      }
+    }
+  }
```

Comment:
It would be good to write a comment here about what is happening in this loop e.g. what steps are taken inside this function. It would be easier to read it if the steps were commented.